SpockOpaleye/src/UserApi.hs

```diff
@@ -17,49 +17,46 @@ module UserApi
   , deactivateUser
   ) where
 
-import           ApiBase
+import           AppCore
 import           Control.Arrow
 import           Control.Lens
 import           Control.Monad.IO.Class
 import           Data.Maybe
-import           DataTypes
 import           GHC.Int
 import           Opaleye
-import           OpaleyeDef
 
-import           CryptoDef
 import           Prelude                    hiding (id)
 
-createUser :: UserIncoming -> AppM (Auditable User)
+createUser :: UserIncoming -> AppM User
 createUser user = do
   Just hash <- liftIO $ bcryptPassword $ user ^. password
   let fullUser = user { _userpolyPassword = hash }
-  auditable <$> (createRow userTable fullUser)
+  createRow userTable fullUser
 
-updateUser :: Auditable User -> AppM (Auditable User)
+updateUser :: User -> AppM User
 updateUser user = updateAuditableRow userTable user
 
-activateUser :: Auditable User -> AppM (Auditable User)
+activateUser :: User -> AppM User
 activateUser user = setUserStatus user UserStatusActive
 
-deactivateUser :: Auditable User -> AppM (Auditable User)
+deactivateUser :: User -> AppM User
 deactivateUser user = setUserStatus user UserStatusInActive
 
-setUserStatus :: Auditable User -> UserStatus -> AppM (Auditable User)
+setUserStatus :: User -> UserStatus -> AppM User
 setUserStatus user newStatus = updateUser $ user & status .~ newStatus
 
-removeUser :: Auditable User -> AppM GHC.Int.Int64
-removeUser Auditable { _data = rUser} = removeRow userTable rUser
+removeUser :: User -> AppM GHC.Int.Int64
+removeUser user = removeAuditableRow userTable user
 
-readUsers :: AppM [Auditable User]
-readUsers = wrapAuditable $ readRow userQuery
+readUsers :: AppM [User]
+readUsers = readRow userQuery
 
-readUsersForTenant :: TenantId -> AppM [Auditable User]
-readUsersForTenant tenantId = wrapAuditable $ readRow $ userQueryByTenantid tenantId
+readUsersForTenant :: TenantId -> AppM [User]
+readUsersForTenant tenantId = readRow $ userQueryByTenantid tenantId
 
-readUserById :: UserId -> AppM (Maybe (Auditable User))
+readUserById :: UserId -> AppM (Maybe User)
```

Comment:
I think this should throw an error if the user is not found. It is highly unlikely that a known UserId is missing from the DB. Think of the usage of this function -- what will the call-site really do if it gets a `Nothing` from this function?